### ISSUE TITLE
Feature request: provide way to set up decayDays in linting config

### ISSUE THREAD

**Issue by MelSumner** [ember-template-lint/ember-template-lint#1686]
I would like to be able to set up decay days in the `.template-lintrc.js` file in bulk. This is helpful when I already have a set list of warn and error decay days for a ruleset. 

For example, a11y rules typically have a SLA due to legal liability. While the number of days to fix may vary by org, and therefore should be set up manually per app, I would like to be able to do something like this:

```js
lintTodo: [
  {
    "rule": "link-href-attributes",
    "decayDays": [
      "warn": 10,
      "error": 21
    ]
  },
 {
    "rule": "no-duplicate-id",
    "decayDays": [
      "warn": 3,
      "error": 7
    ]
  },
 {
    "rule": "no-nested-interactive",
    "decayDays": [
      "warn": 1,
      "error": 2
    ]
  },
]
```

Even more convenient would be if `rule` accepted an array, so rules with the same decayDays could be grouped together:

```js
{
  "rules": [
    "link-href-attributes", 
    "no-duplicate-landmark-elements", 
    "no-heading-inside-button", 
    "no-invalid-role"
  ],
  "decayDays": [
    "warn": 10,
    "error": 21
  ]
}

**Comment by scalvert** [comment]
This is an interesting idea. 

The reason we didn't implement TODO configurations in the `.template-lintrc.js` file is that TODOs are implemented similarly between `ember-template-lint`'s implementation and `@scalvert/eslint-formatter-todo`. As such, we intentionally developed this so that you could share the configurations between both. 

It's certainly an interesting idea to be able to do that. I'm also open to extending this behavior if we think it makes sense. It would feel a bit odd to me to have the global defaults contained in the `package.json`, while other overrides exist in `.template-lintrc.js`. WDYT?

**Comment by rwjblue** [comment]
Hmm, I'm not sure we want to allow such a complicated configuration as this. This complexity doesn't obviously improve things over just always using the lowest range of values.

**Comment by MelSumner** [comment]
> Hmm, I'm not sure we want to allow such a complicated configuration as this. This complexity doesn't obviously improve things over just always using the lowest range of values.

Can you say more? 
- I pulled inspo from how the decayDays can currently be set in the package.json and from the way pending used to work. 
- a11y rules are a little more complicated because they can have a specific number of days to fix them based on the impact to the user, so it would be nice to have this easily supported in the linter itself.

**Comment by MelSumner** [comment]
> This is an interesting idea.
> 
> The reason we didn't implement TODO configurations in the `.template-lintrc.js` file is that TODOs are implemented similarly between `ember-template-lint`'s implementation and `@scalvert/eslint-formatter-todo`. As such, we intentionally developed this so that you could share the configurations between both.
> 
> It's certainly an interesting idea to be able to do that. I'm also open to extending this behavior if we think it makes sense. It would feel a bit odd to me to have the global defaults contained in the `package.json`, while other overrides exist in `.template-lintrc.js`. WDYT?

It might seem a _little_ odd, but I would still expect to have linting configurations in the linting config file. But if we feel like that causes too much dissonance, then no worries. I think I care more about being able to do the thing, rather than where the config goes, if that makes sense.

**Comment by rwjblue** [comment]
> a11y rules are a little more complicated because they can have a specific number of days to fix them based on the impact to the user

Sure, but what I'm saying is that just choosing the lowest of the settings to use all the time is probably perfectly fine.

**Comment by scalvert** [comment]
@rwjblue I think what @MelSumner is referring to is there _may_ be policies that require certain rules to take a higher precedence than others. Take accessibility, for instance. It may be true that certain rule violations violate certain legally required accessibility requirements, and as such we want those rules to have an earlier date set to warn and subsequently error. Being able to do this would be quite helpful.

The question really is, should we add that level of complexity directly to `ember-template-lint`, or allow folks to employ the node API to build a script to set these rules themselves. My vote would be the latter, in order to minimize complexity.

**Comment by MelSumner** [comment]
> @rwjblue I think what @MelSumner is referring to is there _may_ be policies that require certain rules to take a higher precedence than others. Take accessibility, for instance. It may be true that certain rule violations violate certain legally required accessibility requirements, and as such we want those rules to have an earlier date set to warn and subsequently error. Being able to do this would be quite helpful.

Yes, this is really what I mean. I would also want to have strong defaults for this, if for no other reason because it's what our community is used to; it would also make it easier to write about & share common configs.

**Comment by lifeart** [comment]
Moving issue to umbrella ticket https://github.com/ember-template-lint/ember-template-lint/issues/new and closing it here (for now)
